emission/analysis/modelling/user_label_model/run_model.py

```diff
@@ -0,0 +1,201 @@
+        model_data = load(user_id, model_name, model_storage)
+        model.from_dict(model_data)
+        
+        # todo: get timestamp from pipeline, use as filter in query
+        start_ts = -1
+        end_ts = datetime.now()
+        time_query = TimeQuery(
+            timeType="data.start_ts",
+            startTs=start_ts,
+            endTs=end_ts
+        )
+    else:
+        time_query = None
+
+    trips = _get_trips_for_user(user_id, time_query, min_trips)
+
+    # fit and store the updated model
+    model.fit(trips)
+    model_data = model.to_dict()
+    save(user_id, model_data, model_name, model_storage)
+
+    if model.is_incremental:
+        new_timestamp = datetime.now()
+        # todo: update pipeline with new timestamp
+
+    logging.debug(f"{model_name} label prediction model built for user {user_id}")
+
+
+def predict_labels_with_n(
+    trip: Confirmedtrip,
+    model_type = ModelType.GREEDY_SIMILARITY_BINNING,
+    model_storage = ModelStorage.FILE_SYSTEM):
+    """
+    invoke the user label prediction model to predict labels for a trip.
+
+    :param trip: the trip to predict labels for
+    :type trip: Confirmedtrip
+    :param model_type: type of prediction model to run
+    :type model_type: ModelType
+    :param model_storage: location to read/write models
+    :type model_storage: ModelStorage
+    :return: a list of predictions
+    :rtype: List[Prediction]
+    """
+    user_id = trip['user_id']
+    model = _load_user_label_model(user_id, model_type, model_storage)
+    predictions, n = model.predict(trip)
+    return predictions, n
+
+
+def _get_trips_for_user(user_id, time_query: Optional[TimeQuery]=None, min_trips: int=14):
+    """
+    load the labeled trip data for this user, subject to a time query. if the user
+    does not have at least $min_trips trips with labels, then return an empty list.
+
+    :param user_id: user to collect trips from
+    :type user_id: _type_
+    :param time_query: query to restrict the time (optional)
+    :type time_query: Optional[TimeQuery]
+    :param min_trips: minimum number of labeled trips required to train
+    :type min_trips: int
+    """
+    trips = esda.get_entries(key=esda.CONFIRMED_TRIP_KEY, user_id=user_id, time_query=time_query)
+    labeled_trips = [trip for trip in trips if trip['data']['user_input'] != {}]
+    if not len(labeled_trips) >= min_trips:
+        msg = (
+            f"Total: {len(trips)}, labeled: {len(labeled_trips)}, user "
+            f"{user_id} doesn't have enough valid trips for further analysis."
+        )
+        logging.debug(msg)
+        return []
+    return labeled_trips
+
+
+def _load_user_label_model(user_id, model_type: ModelType, model_storage: ModelStorage) -> UserLabelPredictionModel:
+    model = _model_factory(model_type)
+    model_name = model_type.model_name
+
+    model_data = load(user_id, model_name, model_storage)
+    model.from_dict(model_data)
+
+    return model
+
+
+if __name__ == '__main__':
```

Comment:
oh, these were the test cases at the bottom of the `tour_model_first_only/load_predict` file, though i'm not sure what dataset they were based on. at least not the dataset documented [here](https://github.com/e-mission/e-mission-server#loading-test-data)